holopy/scattering/tests/test_calculations.py

```diff
@@ -30,65 +31,128 @@
 from holopy.core import detector_grid
 from holopy.core.tests.common import assert_obj_close
 from holopy.scattering.calculations import *
+from holopy.scattering.errors import MissingParameter
 
-scatterer = Sphere(n=1.6, r=.5, center=(5, 5, 5))
-medium_index = 1.33
-locations = detector_grid(shape=(20, 20), spacing=.1)
-wavelen = 0.66
-polarization=(0, 1)
+import xarray as xr
 
-@attr("fast")
-def test_calc_holo():
-    holo = calc_holo(locations, scatterer, medium_index, wavelen, polarization)
+SCATTERER = Sphere(n=1.6, r=.5, center=(5, 5, 5))
+MED_INDEX = 1.33
+LOCATIONS = detector_grid(shape=(20, 20), spacing=.1)
+WAVELEN = 0.66
+POL = (0, 1)
 
 
-def test_calc_field():
-    field = calc_field(locations, scatterer, medium_index, wavelen, polarization)
+class TestCalculations(unittest.TestCase):
+    @attr("fast")
+    def test_calc_holo(self):
+        holo = calc_holo(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
+    def test_calc_field(self):
+        field = calc_field(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
-@attr("fast")
-def test_calc_cross_section():
-    cross = calc_cross_sections(scatterer, medium_index, wavelen, polarization)
+    @attr("fast")
+    def test_calc_cross_sections(self):
+        cross = calc_cross_sections(SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
+    def test_calc_intensity(self):
+        intensity = calc_intensity(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN, POL)
+        self.assertTrue(True)
 
-def test_calc_intensity():
-    intensity = calc_intensity(locations, scatterer, medium_index, wavelen, polarization)
+    @attr("fast")
+    def test_calc_scat_matrix(self):
+        matr = calc_scat_matrix(LOCATIONS, SCATTERER, MED_INDEX, WAVELEN)
+        self.assertTrue(True)
 
+    def test_finalize(self):
+        detector = finalize(LOCATIONS.values, LOCATIONS)
+        self.assertTrue(True)
```

Comment:
I realize that the old tests just checked that the code didn't crash, but for these tests
(test_finalize, test_calc_scat_matrix, test_calc_intensity, test_calc_cross_sections, test_calc_field, test_calc_holo) it might be good to check something more than that, given that we have the hologram. A few options are to check a particular value (maybe not great, since it might be too stringent if we change the theories), to check the return type (maybe not great if we move away from xarray) or to at least check that the return type is not None (maybe not great because it's not a very stringent test).

Reply:
Addressed in 1d6a33a and 79b6dc6.

For now we will test for specific return values.